pkg/controller/admissionchecks/provisioning/controller_test.go

```diff
@@ -0,0 +1,738 @@
+*/
+
+package provisioning
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	autoscaling "k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/apis/autoscaling.x-k8s.io/v1beta1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/util/slices"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+)
+
+var (
+	wlCmpOptions = []cmp.Option{
+		cmpopts.EquateEmpty(),
+		cmpopts.IgnoreTypes(metav1.ObjectMeta{}, metav1.TypeMeta{}),
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+		cmpopts.IgnoreFields(kueue.AdmissionCheckState{}, "LastTransitionTime"),
+	}
+
+	reqCmpOptions = []cmp.Option{
+		cmpopts.EquateEmpty(),
+		cmpopts.IgnoreTypes(metav1.ObjectMeta{}, metav1.TypeMeta{}),
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+	}
+
+	tmplCmpOptions = []cmp.Option{
+		cmpopts.EquateEmpty(),
+		cmpopts.IgnoreTypes(metav1.ObjectMeta{}, metav1.TypeMeta{}),
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+		cmpopts.IgnoreFields(corev1.PodSpec{}, "RestartPolicy"),
+	}
+
+	acCmpOptions = []cmp.Option{
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+	}
+)
+
+func requestWithCondition(r *autoscaling.ProvisioningRequest, conditionType string, status metav1.ConditionStatus) *autoscaling.ProvisioningRequest {
+	r = r.DeepCopy()
+	apimeta.SetStatusCondition(&r.Status.Conditions, metav1.Condition{
+		Type:   conditionType,
+		Status: status,
+	})
+	return r
+}
+
+func TestReconcile(t *testing.T) {
+	baseWorkload := utiltesting.MakeWorkload("wl", TestNamespace).
+		PodSets(
+			*utiltesting.MakePodSet("ps1", 4).
+				Request(corev1.ResourceCPU, "1").
+				Obj(),
+			*utiltesting.MakePodSet("ps2", 4).
+				Request(corev1.ResourceMemory, "1M").
+				Obj(),
+		).
+		ReserveQuota(utiltesting.MakeAdmission("q1").PodSets(
+			kueue.PodSetAssignment{
+				Name: "ps1",
+				Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+					corev1.ResourceCPU: "flv1",
+				},
+				ResourceUsage: map[corev1.ResourceName]resource.Quantity{
+					corev1.ResourceCPU: resource.MustParse("4"),
+				},
+				Count: ptr.To[int32](4),
+			},
+			kueue.PodSetAssignment{
+				Name: "ps2",
+				Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+					corev1.ResourceCPU: "flv2",
+				},
+				ResourceUsage: map[corev1.ResourceName]resource.Quantity{
+					corev1.ResourceCPU: resource.MustParse("3M"),
+				},
+				Count: ptr.To[int32](3),
+			},
+		).
+			Obj()).
+		AdmissionChecks(kueue.AdmissionCheckState{
+			Name:  "check1",
+			State: kueue.CheckStatePending,
```

Comment:
Let's add an Admission check that is not related to ProvReq

Reply:
done